### PR TITLE
Extend bindings for OBB and misc fixes.

### PR DESCRIPTION
**Pull request by Skylion007**
## Motivation and Context

Added some functionality needed for properly using OBBs in the Habitat-API. 
1). Add new python bindings to OBB
2). Fix bugs computeMOBB and expose to python.
3). Have sampleNavigablePoint return a point that will always not be navigable if it fails. That ensure that we get a proper error state that we can check against pt(inf,inf,inf).
## How Has This Been Tested

It has been tested by being used for generating the ObjectNav datasets in the Habitat-API.
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: src/esp/nav/PathFinder.cpp [hunk 1]
```diff
@@ -854,7 +854,8 @@ static float frand() {
 
 vec3f PathFinder::Impl::getRandomNavigablePoint() {
   dtPolyRef ref;
-  vec3f pt;
+  float inf = std::numeric_limits<float>::infinity();
```
Comment (erikwijmans) [inline src/esp/nav/PathFinder.cpp]:
```suggestion
  constexpr float inf = std::numeric_limits<float>::infinity();
```

File: src/esp/geo/OBB.cpp [hunk 2]
```diff
@@ -107,7 +107,7 @@ OBB computeGravityAlignedMOBB(const vec3f& gravity,
     const float dd = d0[0] * d1[1] - d0[1] * d1[0];
 
     const float dx = s1[0] - s0[0];
-    const float dy = s1[1] - s0[0];
+    const float dy = s1[1] - s0[1];
```
Comment (mathfac) [inline src/esp/geo/OBB.cpp]: @Skylion007, can you remind me what was the bug here?

File: src/esp/geo/OBB.cpp [hunk 3]
```diff
@@ -240,7 +240,7 @@ OBB computeGravityAlignedMOBB(const vec3f& gravity,
     aabb.extend(T_w2b * pt);
   }
 
-  return OBB{aabb.center(), aabb.sizes(), T_w2b.inverse()};
+  return OBB{T_w2b.inverse() * aabb.center(), aabb.sizes(), T_w2b.inverse()};
```
Comment (mathfac) [inline src/esp/geo/OBB.cpp]: And here.

File: src/esp/geo/geo.cpp [hunk 4]
```diff
@@ -39,13 +39,12 @@ std::vector<vec2f> convexHull2D(const std::vector<vec2f>& points) {
   }
 
   // Build upper hull
-  for (size_t i = idx.size() - 2, t = k + 1; i >= 0; i--) {
```
Comment (mathfac) [inline src/esp/geo/geo.cpp]: And here, just to have a history record for all this deep changes.

File: src/esp/geo/OBB.cpp (hunk 2 above)
Comment (Skylion007) [reply src/esp/geo/OBB.cpp]: dy was not actually calculating a delta y.

File: src/esp/geo/OBB.cpp (hunk 3 above)
Comment (Skylion007) [reply src/esp/geo/OBB.cpp]: The center was not in world coordinates

File: src/esp/geo/geo.cpp (hunk 4 above)
Comment (Skylion007) [reply src/esp/geo/geo.cpp]: Resolves a segfault that occurs on Linux due to undefined behavior of unsigned integer underflows.

**Pull request merged**